packages/atlas/src/components/_crt/CreateTokenDrawer/steps/SetupTokenStep.tsx

```diff
@@ -0,0 +1,123 @@
+import { OptionCardGroupRadio } from '@/components/_inputs/OptionCardGroup'
+import { RatioSlider } from '@/components/_inputs/Slider'
+import { useMountEffect } from '@/hooks/useMountEffect'
+
+import { CommonStepProps } from './types'
+
+const accessOptions = [
+  {
+    label: 'Anyone',
+    caption: 'Everyone can own your token.',
+    icon: <SvgActionUnlocked />,
+    value: true,
+  },
+  {
+    label: 'Invite only',
+    caption: 'Only members on allowlist can own your token. ',
+    icon: <SvgActionLock />,
+    value: false,
+  },
+]
+
+type SetupTokenStepProps = {
+  onSubmit: (form: SetupStepForm) => void
+} & CommonStepProps
+
+export const SetupTokenStep = ({ setPrimaryButtonProps, onSubmit, form }: SetupTokenStepProps) => {
+  const {
+    register,
+    control,
+    handleSubmit,
+    formState: { errors },
+  } = useForm<SetupStepForm>({
+    defaultValues: form,
+  })
+
+  useMountEffect(() => {
+    setPrimaryButtonProps({
+      text: 'Next step',
+      onClick: () => handleSubmit(onSubmit)(),
+    })
+  })
+
+  return (
+    <CrtFormWrapper title="Set up your token" subtitle="Enter basic token information and settings." learnMoreLink="">
+      <FormField
+        label="Name"
+        description="Choose 3 letter name for your token to be displayed on your token page, tokens marketplace and in your buyers’ portfolio."
+        error={errors.name?.message}
+      >
+        <Input
+          {...register('name', {
+            maxLength: {
+              value: 3,
+              message: 'Name must be 3 letters long.',
+            },
+            minLength: {
+              value: 3,
+              message: 'Name must be 3 letters long.',
+            },
+            required: {
+              value: true,
+              message: 'Name is required.',
+            },
+            pattern: {
+              value: /[A-Z]+/,
+              message: 'Name should contain only uppercase letters.',
+            },
+          })}
+          placeholder="ABC"
+          nodeStart={
+            <Text variant="t300" as="p" color="colorTextMuted">
+              $
+            </Text>
+          }
+        />
+      </FormField>
+      <FormField label="Access" description="Define if everyone can buy your token or only selected memebers.">
+        <Controller
+          name="isOpen"
+          control={control}
+          render={({ field: { value, onChange } }) => (
+            <OptionCardGroupRadio options={accessOptions} onChange={onChange} selectedValue={value} />
+          )}
+        />
+      </FormField>
+      <FormField
+        label="Revenue share with holders"
+        description="Define the share of your channel revenue that goes to yourself vs shared with your token holders.
+Recommended values — Channel: 80%, Holders:20%. "
+      >
+        <Controller
+          name="revenueShare"
+          control={control}
+          render={({ field }) => <RatioSlider id="revenueShare" {...field} />}
+        />
```

Comment:
I don't remember if I tested it but since `RatioSlider` can work as an uncontrolled component doesn't it work without being wrapped in `<Controller ...` ?
```suggestion
        <RatioSlider {...register('revenueShare')} id="revenueShare" />
```

Reply:
@WRadoslaw actually that probably won't work since the input itself is controlled